cmd/kubeadm/app/cmd/join.go

```diff
@@ -116,6 +116,13 @@ func NewCmdJoin(out io.Writer) *cobra.Command {
 	cmd.PersistentFlags().StringVar(
 		&cfg.TLSBootstrapToken, "tls-bootstrap-token", "",
 		"A token used for TLS bootstrapping")
+	cmd.PersistentFlags().StringSliceVar(
+		&cfg.TLSDiscoveryRootCAPubKeys, "tls-discovery-root-ca", []string{},
+		"Validate that the root CA public key matches this SHA-256 SPKI hash (repeat to allow multiple public keys).")
```

Comment:
Flag doc and name should indicate that this is only relevant for token discovery.